### PR TITLE
[Fix](bangc-ops) get num_act_out from sparse_conv_desc

### PR DESCRIPTION
**Pull request by Devin-D-U**
Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

Please describe your motivation and the goal you want to achieve through this pull request.

## 2. Modification

Please briefly describe what modification is made in this pull request, and indicate where to make the modification.

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS Accuracy Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- [ ] diff1: diff1 <= 3e-3
- [ ] diff2: diff2 <= 3e-3

#### 3.1.2 Operator Scheme checklist

|     No.        |                 Details              |            Check Results             |
|----------------|--------------------------------------|--------------------------------------|
|        1       |Supported hardware                    |             MLU370<br>MLU590         |
|        2       |Job types                             |          block <br> U1 <br> U4       |
|        3       |Layouts                               |          NHWC 、NCHW、ARRAY etc      |
|        4       |Whether multi-dimensions are supported|                                      |
|        5       |Whether element zero is supported     |                                      |
|        6       |Data type(half/float)                 |           half / float etc           |
|        7       |Whether there is size limit           |                                      |

#### 3.1.3 New Feature Test

If you have checked the following items, please tick the relevant box.

- [ ] Data type test
- [ ] Multidimensional tensor test
- [ ] Layout test
- [ ] Different size/integer remainder end segment/alignment misalignment test
- [ ] Zero dimensional tensor test/zero element test
- [ ] stability test
- [ ] Multiple platform test
- [ ] Gen_case module test
- [ ] Nan/INF tests 
- [ ] Bug fix tests
- [ ] For memory leak check details, see[GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For code coverage check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For I/O calculation efficiency check details, see: [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md).

#### 3.1.4 Parameter Check

When a new operator is submitted, the test points are given and the test results are stated.

|                   Test Point                    | Acceptance Standard | Test Result (Error Message) |
| ----------------------------------------------- | --------------------| --------------------------- |
| Whether it conforms to the operator restriction |     Normal error    |                             |
| Whether illegal parameters are passed           |     Normal error    |                             |

### 3.2 Accuracy Test

For the cases used in the New Feature Test section, the features and the number of cases are recorded here. When multiple operations are tested, multiple tables are needed to include details of these operations.

Operation:

|Test Point           | Description                      | Quantity |  Comment |
|----------           |----------------------------------|----------|  --------|
|Data type test       |half/float/int8                   |          |          |
|Mult-tensor test     |Supports 1-8 dims                 |          |          |
|Layout test          |Supports NCHW/NHWC                |          |          |
|Zero element test    |Whether to support this test      |          |          |
|Stability test       |--gtest_repeat=NUM<br>--thread=NUM|          |          |
|Mult-platform test   |MLU370/MLU590                     |          |          |
|Nan/INF test         |Whether to support this test      |          |          |

MLU Hardware Time      ]: 118 (us)
[MLU Interface Time     ]: 119.534 (us)
[MLU IO Efficiency      ]: 0.0093031
[MLU Compute Efficiency ]: 0.0136398
[MLU Workspace Size     ]: 5.71098e+06 (Bytes)
[MLU TheoryOps          ]: 3.70829e+06 (Ops)
[MLU TheoryIOs          ]: 3.0351e+06 (Bytes)
[MLU ComputeForce       ]: 2.304e+12 (op/s)
[MLU IoBandWidth        ]: 2764.8 (GB/s)
[GPU Hardware Time      ]: -1 (us)
[GPU IO Efficiency      ]: -1
[GPU Compute Efficiency ]: -1
[GPU Workspace Size     ]: -1 (Bytes)
[Diffs]:
[output1]
DIFF3: 0.000000e+00
[output2]
DIFF3: 0.000000e+00
[output3]
DIFF3: 0.000000e+00
[^      OK ] /home/get_indice_pairs/get_indice_pairs_data_included_int32_1675738558214.pb
[       OK ] get_indice_pairs/TestSuite.mluOp/201 (13 ms)
[----------] 202 tests from get_indice_pairs/TestSuite (87119 ms total)

[----------] Global test environment tear-down
[2023-2-16 18:17:28] [MLUOP] [Vlog]:TearDown CNRT environment.
[ SUMMARY  ] Total 202 cases of 1 op(s).
ALL PASSED.
[==========] 202 test cases from 1 test suite ran. (87469 ms total)
[  PASSED  ] 202 test cases.

### 3.3 Performance Test

See [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform：MLU370

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|-----|----|----|----|-----|
|op_name|    |    |     |    |    |    |     |
|op_name|    |    |     |    |    |    |     |

Platform：MLU590

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|----|----|----|----|-----|
|op_name|    |    |    |    |    |    |     |
|op_name|    |    |    |    |    |    |     |

### 3.4 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

File: bangc-ops/kernels/get_indice_pairs/get_indice_pairs_structs.cpp [hunk 1]
```diff
@@ -110,6 +110,15 @@ mluOpStatus_t MLUOP_WIN_API mluOpSetSparseConvolutionDescriptor(
   return MLUOP_STATUS_SUCCESS;
 }
 
+mluOpStatus_t MLUOP_WIN_API mluOpGetSparseConvolutionNumActOut(
+    mluOpSparseConvolutionDescriptor_t desc,
+    int *num_act_out) {
+    int size = 0;
+    size = desc->num_act_out;
+    num_act_out[0] =  size;
+    return MLUOP_STATUS_SUCCESS;
```
Comment (Unireverse) [inline bangc-ops/kernels/get_indice_pairs/get_indice_pairs_structs.cpp:119]: 缩进问题

File: bangc-ops/kernels/get_indice_pairs/get_indice_pairs_structs.cpp [hunk 2]
```diff
@@ -110,6 +110,15 @@ mluOpStatus_t MLUOP_WIN_API mluOpSetSparseConvolutionDescriptor(
   return MLUOP_STATUS_SUCCESS;
 }
 
+mluOpStatus_t MLUOP_WIN_API mluOpGetSparseConvolutionNumActOut(
+    mluOpSparseConvolutionDescriptor_t desc,
```
Comment (alex-xuwenming) [inline bangc-ops/kernels/get_indice_pairs/get_indice_pairs_structs.cpp:114]: 要做指针非空检查。

File: bangc-ops/kernels/get_indice_pairs/get_indice_pairs_structs.cpp [hunk 3]
```diff
@@ -110,6 +110,15 @@ mluOpStatus_t MLUOP_WIN_API mluOpSetSparseConvolutionDescriptor(
...
+    int *num_act_out) {
+    int size = 0;
```
Comment (duzekunKTH) [inline bangc-ops/kernels/get_indice_pairs/get_indice_pairs_structs.cpp:116]: 需要对 desc和num_act_out做 空指针防呆

File: bangc-ops/mlu_op.h [hunk 4]
```diff
@@ -1055,6 +1055,32 @@ mluOpSetSparseConvolutionDescriptor(mluOpSparseConvolutionDescriptor_t desc,
                                     const int transpose,
                                     const int inverse);
 
+// Group:: GetIndicePairs
+/*!
+ *  @brief The parameter num_act_out will be obtained from the mluOpSparseConvolutionStruct.
```
Comment (Unireverse) [inline bangc-ops/mlu_op.h:1060]: Obtain num_act_out from mluOpSparseConvolutionStruct. This interface should be called after ::mluOpGetIndicePairs.

**Review by Unireverse — APPROVED** [review]

**Review by alex-xuwenming — APPROVED** [review]

**Review by duzekunKTH — APPROVED** [review]

File: bangc-ops/mlu_op.h [hunk 5]
```diff
@@ -1055,6 +1055,31 @@ mluOpSetSparseConvolutionDescriptor(mluOpSparseConvolutionDescriptor_t desc,
...
                                     const int inverse);
 
+// Group:: GetIndicePairs
+/*!
+ *  @brief Obtains the parameter num_act_out from ::mluOpSparseConvolutionStruct.
+ *
+ *  @param[in] desc
+ *  Pointer to the parameter num_act_out that holds information about the tensor descriptor.
+ *  @param[out] num_act_out
+ *  The parameter num_act_out will be obtained from the ::mluOpSparseConvolutionStruct.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:1065]: 这里写一下这个参数得含义吧

File: bangc-ops/mlu_op.h [hunk 6]
```diff
@@ -6896,6 +6921,7 @@ mluOpScatterNd_v2(mluOpHandle_t handle,
  *
  * @note
  * - This function only supports on MLU300 series or above platforms.
+ * - The parameter num_act_out will be obtained from the mluOpSparseConvolutionStruct.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:6924]: from ::mluOpSparseConvolutionStruct.

File: bangc-ops/mlu_op.h [hunk 7]
```diff
@@ -6896,6 +6921,7 @@ mluOpScatterNd_v2(mluOpHandle_t handle,
  *
  * @note
  * - This function only supports on MLU300 series or above platforms.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:6923]: This function is only supported on MLU300 series or above platforms.  这句顺便帮改一下吧

**Review by GraceGuanH — COMMENTED** [review]
Please check the comments.

**Pull request merged**